lib/OpenQA/Task/Git/Clone.pm

```diff
@@ -97,7 +97,11 @@ sub _git_clone ($app, $job, $ctx, $path, $url) {
 
     return if ($requested_branch and $requested_branch !~ tr/a-f0-9//c and $git->check_sha($requested_branch));
 
-    die "NOT updating dirty git checkout at $path" if !$git->is_workdir_clean();
+    die <<~"END_OF_ERROR_MESSAGE" unless $git->is_workdir_clean;
```

Comment:
I think one problem might still be that this error will not appear in the logfile, only as the reason for the failed minion job. At least the original question was about the logfile not printing a helpful message